packages/eslint-plugin/src/rules/no-shadow.ts

```diff
@@ -566,6 +566,28 @@ export default createRule<Options, MessageIds>({
       };
     }
 
+    /**
+     * Checks if the initialization of a variable has the declare modifier in a
+     * definition file.
+     * @param variable The variable to check.
+     * @returns Whether or not the initialization of a variable has the declare
+     * modifier in a definition file.
```

Comment:
[Docs] Nit: I don't think these JSDocs add anything. From just the name we know this is checking _"is declare in dts file?"_ for a _"variable"_. The first brief description is the only piece that adds new info IMO.

```suggestion
```

```suggestion
```